### PR TITLE
Add team view

### PR DESCRIPTION
**Pull request by tanguoyang**
Add teamlist panel to show existing teams
Add team stats to show aggregates of each team, including unique tag and techstack count

### CODE REVIEW

File: src/main/java/seedu/address/logic/LogicManager.java [hunk 1]
```diff
@@ -71,6 +73,9 @@ public ObservableList<Contact> getFilteredContactList() {
         return model.getFilteredContactList();
     }
 
+    @Override
+    public ObservableList<Team> getTeamList() {return model.getCodeConnect().getTeamList();}
```
Comment (angsongyee) [inline src/main/java/seedu/address/logic/LogicManager.java:77]: You should add line breaks to the body of this method.
```suggestion
    public ObservableList<Team> getTeamList() {
        return model.getCodeConnect().getTeamList();
    }
```

File: src/main/java/seedu/address/model/team/TeamStats.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import java.util.*;
+
+public class TeamStats {
+
+    public final Map<TechStack, Integer> techStack;
+
+    public final Map<Tag, Integer> tags;
+
+    public final Integer size;
```
Comment (angsongyee) [inline src/main/java/seedu/address/model/team/TeamStats.java:15]: Class fields should be declared as private.

File: src/main/java/seedu/address/model/team/TeamStats.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package seedu.address.model.team;
+
+import seedu.address.model.contact.Contact;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.techstack.TechStack;
+
+import java.util.*;
+
+public class TeamStats {
```
Comment (angsongyee) [inline src/main/java/seedu/address/model/team/TeamStats.java:9]: This class should have a JavaDoc too.

File: src/main/java/seedu/address/model/team/TeamStats.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
+package seedu.address.model.team;
+
+import seedu.address.model.contact.Contact;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.techstack.TechStack;
+
+import java.util.*;
```
Comment (angsongyee) [inline src/main/java/seedu/address/model/team/TeamStats.java:7]: Wildcard import

File: src/main/java/seedu/address/model/team/Team.java [hunk 5]
```diff
@@ -17,12 +17,16 @@ public class Team {
     private final Name name;
     private final UniqueContactList members = new UniqueContactList();
 
+    public final TeamStats stats;
```
Comment (angsongyee) [inline src/main/java/seedu/address/model/team/Team.java:20]: Class fields should be declared as private.

**Review by angsongyee — CHANGES_REQUESTED** [review]

**Review by angsongyee — APPROVED** [review]

**Review by Teee728 — APPROVED** [review]
LGTM

**Pull request merged**